### DISCUSSION TITLE
Gõ ký tự toán học ∑ bằng chức năng gõ tắt

### DISCUSSION THREAD

**Discussion by cutbypham** [BambooEngine/ibus-bamboo#282] [Show and tell]
- Khi tui muốn gõ mấy cái ký tự của toán thì phải lên mạng search cái ký tự của nó rồi copy paste lại giống như ^2 = ²
- Bí một chỗ nữa là mấy cái ký tự toán thường ngày viết trên giấy okay, nhưng khi gõ trên máy tính thì méo biến tên của nó là gì thường mấy cái unicode của toán thì phải search tiếng anh mới ra, mà tui mới biết mấy cái tên cơ bản thư alpha symbol -> α, β, lấy ví dụ như mấy cái symbol như sum: ∑ rồi ℝ thì hên xui lúc nhớ lúc không
- Nói về cá nhân tui thì sợ viết trên giấy lém, tui được mọi người ca ngợi là sỡ hữu chữ viết bác sĩ, viết giống thư pháp các thể loại 😁, nên tui thường note trên laptop, phone là chính còn giấy dùng khi nào tui phải suy nghĩ cái vấn đề nào đó cần phải tái hiện cái suy nghĩ của tui.

**⇒ Okay đến lúc giải quyết vấn đề**

- Thì tui xài linux nên sử dụng cái ibus-bamboo[^1] để gõ tiếng việt, còn mấy cái trình gõ tiếng việt khác thì tui nghĩ cũng cùng ý tưởng.
- Tui dùng chức năng gõ tắt của ibus-bamboo, nói chung thì gõ một cái chứ gì đó như `r..` đi thì nó bung ra ℝ, cơ bản ý tưởng là vậy.

ví dụ một khúc file cấu hình:
```
a..:α
b..:β
z..:ℤ
r..:ℝ
p..:π
<=..:≤
=>..:⇒
->..:→
<-..:←
^2..:²
^3..:³
^4..:⁴
```

⇒ Thì ở đây tui dùng leader key là `..`, để chắc là tui muốn thay thế ký tự, mấy bạn có thể không sử dụng leader key cũng được.

> ví dụ: giống như bạn muốn gõ ^5 mà bạn cấu hình ^5:⁵, thì bạn có thể bấm phím tắt chuyển sang tiếng anh để gõ cũng là một cách.

- Tui lấy mấy cái tưởng này từ chú Xah Lee[^2], tui coi cái emacs math input[^3] của chú ý mà cái math input đó mấy trăm cái symbol lận, trình toán của tui cũng không tới đẳng cấp sau khi học xong tiến sĩ về toán của chú ý nên tui coi rồi lụm mấy cái tui thường thấy thôi ⇒ nói chung là do tui lười ý 😁.
- Còn về giới hạn của cái này thì bản chất nó là unicode nên là từng ký tự một nên không làm mấy cái xịn hơn như lim, rồi phần tử phần mẫu 2-tử-/3-mẫu-, rồi giới hạn,... tui tui nghĩ là làm được nhưng nó mệt lém giống như tụi mình sẽ thiết kế một cái ngôn ngữ mới vậy nên do trình không đủ nên khó làm lắm, nếu mấy bạn muốn mấy cái siêu đẳng chuyên về nghiên cứu về toán học thì có thể tìm hiểu wolfram lang[^4] nó xịn lém.

⇒ Tóm lại thì tui sẽ tiếp tục thêm bớt, nên sẽ tạo một cái repo ibus-bamboo-math[^5] trên github rồi đi chém gió trên ibus-bamboo, có gì vô repo đó[^5] hay chat với tui để để dàm đạo hỉ.

---

[^1]: https://github.com/BambooEngine/ibus-bamboo/
[^2]: http://xahlee.info/
[^3]: https://github.com/emacsmirror/xah-math-input/blob/master/xah-math-input.el
[^4]: http://xahlee.info/M/index.html
[^5]: https://github.com/thuanpham2311/ibus-bamboo-math

**Comment by cutbypham** [comment]
- Có gì vô đọc cái file text đàm đạo với mọi người hỉ: https://github.com/thuanpham2311/ibus-bamboo-math
- Nói chung mọi người cứ gửi pull request thổi mái nha 😁

**Comment by cutbypham** [comment]
sao 1 tuần chiêm ngẫm thì những ký tự thường dùng là: ⇒ → • ² ≥

**Reply by cutbypham** [reply]
cũng bõ .. mới hiện ra được, không xài `..` cũng không bị conflict nhiều nên bỏ  `..` gõ cho khỏe :smile:

**Comment by t1ld3x** [comment]
bạn nghĩ sao nếu tích hợp bảng mã unicode vào tính năng gõ Emoji rồi gợi ý các kí tự unicode/toán học giống như gợi ý emoji 🤔

**Reply by cutbypham** [reply]
- Mình thấy ý này khá hay.
- Mình thì thường gõ emoji bằng cách dùng emoji fzf 
giống như vầy:
![image](https://user-images.githubusercontent.com/40050527/138544097-f1568b87-a55c-471a-b50f-a7d000e15c6f.png)
- Mình thấy dùng cách tìm kiếm của [fzf](https://github.com/junegunn/fzf) là okay nhất, cộng với nhiều cách định nghĩ khác nhau giống hình trên để dễ cho người dùng tìm kiếm.
- thật ra thì dùng cách gõ tắt thì mình chỉ nhớ những ký tự thường xài như là '->' → '=>' ⇒ 'bu' • thôi còn mấy cái không thường xài như 'cuz' ∵ thì có khi quên, nếu tích hợp vào chứng gợi ý thì search nó dễ hơn.
- giống như mình map phím gợi ý là `ctrl-e`, giờ mình chỉ cần bấm `ctrl-e` gõ `bu` nó hiện ra • rồi `enter` cái là nó ra • liền rất okay, tính đâu đó khoảng 5 lần chạm thôi so với `bu..` là khoảng 5 lần tính luôn phím space để thực hiện, nếu `bu` không là khoảng 3 lần nhưng dễ bị conflict các thể loại, còn không nhớ phải mở bảng gõ tắt kiếm thì nó còn tốn sức nhiều hơn.

**⇒ tóm lại nếu thực hiện được chứng năng này thì rất rất okay.**

**Comment by ndgnuh** [comment]
Mình nghĩ nếu làm thành tính năng trong engine nên có một cái phím tắt giống như ctrl shift u, và khi nào nhấn tổ hợp đó thì mới compose ^2 thành ². Nếu làm việc với công thức toán thì tính năng này sẽ khá hữu dụng, nhưng có khả năng bị conflict với những tool khác mà đội làm toán hay dùng (latex hoặc code).
